src/core/DataFile.cpp

```diff
@@ -1916,14 +1916,15 @@ void DataFile::loadData( const QByteArray & _data, const QString & _sourceFile )
 		if( !success ) qWarning("File Version conversion failure.");
 	}
 
+	// Perform upgrade routines
+	if (m_fileVersion < UPGRADE_METHODS.size()) { upgrade(); }
```

Comment:
`upgrade()` sets the `creatorversion` attribute to the current LMMS version, so calling it here means the next block of code can't see the original value of that attribute.

Reply:
Oh right, I didn't realize it touched the XML